### PR TITLE
Describe how to develop an extension using gradle

### PR DESCRIPTION
**Pull request by glefloch**
This branch introduce how to create a quarkus extension based on gradle. I edited the `building-my-first-extension` guide as most of the guide is not related to maven specific things.
I'm not sure we need a specific guide for `building-my-first-extension-with-gradle`. 
WDYT @aloubyansky, @gastaldi ?

### CODE REVIEW

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 1]
```diff
@@ -383,6 +389,103 @@ The key points are:
...
+=== Gradle setup
+
+Quarkus does not provide any way to initialize a gradle project for extension yet.
+
+As mentionned before, an extension is composed of two modules:
+
+* `runtime`
+* `deployment`
+
+We are going to create a gradle multi module project with those two modules. Here is a simple `settings.gradle` example file:
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:401]:
```suggestion
We are going to create a Gradle multi module project with those two modules. Here is a simple `settings.gradle` example file:
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 2]
```diff
@@ -383,6 +389,103 @@ The key points are:
 <3> We add  the `quarkus-bootstrap-maven-plugin` to generate the Quarkus extension descriptor included into the runtime artifact which links it with the corresponding deployment artifact.
 <4> We add  the `quarkus-extension-processor` to the compiler annotation processors.
 
+=== Gradle setup
+
+Quarkus does not provide any way to initialize a gradle project for extension yet.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:394]:
```suggestion
Quarkus does not provide any way to initialize a Gradle project for extension yet.
```

Comment (aloubyansky) [reply docs/src/main/asciidoc/building-my-first-extension.adoc:394]: There is a way to do that using the CLI. `quarkus create --gradle`. You could reference the CLI page.

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 3]
```diff
@@ -383,6 +389,103 @@ The key points are:
...
+<1> Apply the `java-library` plugin for all sub-modules
+<2> Apply the `maven-publish` plugin used to publish our artifacts
+<3> Globally set the group id used for publication
+
+The `io.quarkus.extension` plugin will be used in order to help us building the extension.
+The plugin will *only* be applied to the `runtime` module.
+
+==== The deployment module
+
+The deployment module is does not require any specific plugin.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:445]:
```suggestion
The deployment module does not require any specific plugin.
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 4]
```diff
@@ -87,7 +87,13 @@ So, the more code you run during the two phases above, the faster your applicati
 
 Now that everything is explained, we can start coding!
 
-== Maven setup
+== Project setup
+
+Extensions can be built with either Maven or Gradle. Depending on your build tool, setup can be done as following:
+
+NOTE: The gradle extension plugin is still experimental and may be missing features available in the maven plugin.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:94]:
```suggestion
NOTE: The Gradle extension plugin is still experimental and may be missing features available in the Maven plugin.
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 5]
```diff
@@ -403,6 +506,13 @@ All we need to do is add `quarkus-undertow` as dependency to our `./greeting-ext
     </dependency>
 ----
 
+For gradle, add the dependency in `./greeting-extension/runtime/build.gradle` file:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:509]:
```suggestion
For Gradle, add the dependency in `./greeting-extension/runtime/build.gradle` file:
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 6]
```diff
@@ -501,6 +611,13 @@ Let's add `quarkus-undertow-deployment` as dependency to our `./greeting-extensi
 NOTE: The dependency on `quarkus-arc-deployment` generated by the `create-extension` mojo can now be removed since
 `quarkus-undertow-deployment` already depends on it.
 
+For gradle, add the dependency in `./greeting-extension/deployment/build.gradle` file:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:614]:
```suggestion
For Gradle, add the dependency in `./greeting-extension/deployment/build.gradle` file:
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 7]
```diff
@@ -552,11 +669,20 @@ Let's add the `rest-assured` dependency into the  `./greeting-extension/deployme
     ...
     <dependency>
       <groupId>io.rest-assured</groupId>
-      <artifactId>rest-assured</artifactId> <!--2-->
+      <artifactId>rest-assured</artifactId>
       <scope>test</scope>
     </dependency>
 ----
 
+For gradle, add the dependency in `./greeting-extension/deployment/build.gradle` file:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:677]:
```suggestion
For Gradle, add the dependency in `./greeting-extension/deployment/build.gradle` file:
```

**Comment by glefloch** [comment]
Thanks for your reviews, I updated my branch with fixes.

**Review by aloubyansky — APPROVED** [review]

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 8]
```diff
@@ -383,6 +389,103 @@ The key points are:
 <3> We add  the `quarkus-bootstrap-maven-plugin` to generate the Quarkus extension descriptor included into the runtime artifact which links it with the corresponding deployment artifact.
 <4> We add  the `quarkus-extension-processor` to the compiler annotation processors.
 
+=== Gradle setup
+
+Quarkus does not provide any way to initialize a Gradle project for extension yet.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:394]:
```suggestion
Quarkus does not provide any way to initialize a Gradle project for extensions yet.
```

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 9]
```diff
@@ -383,6 +389,103 @@ The key points are:
...
+
+Here is an example of `build.gradle` file for the `runtime` module:
+
+[source, groovy]
+----
+plugins {
+    id 'io.quarkus.extension' <1>
+}
+
+name = 'greeting-extension' <2>
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:479]: Is there a way to set the description of the extension too?

File: docs/src/main/asciidoc/building-my-first-extension.adoc [hunk 10]
```diff
@@ -383,6 +389,103 @@ The key points are:
...
+
+dependencies {
+    implementation platform("io.quarkus:quarkus-bom:${quarkus.version}")
+}
+----
+
+<1> Apply the `io.quarkus.extension` plugin.
+<2> By convention, the runtime module has no suffix (`greeting-extension`) as it is the artifact exposed to the end user
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-my-first-extension.adoc:487]:
```suggestion
<2> By convention, the runtime module doesn't have a suffix (and thus is named `greeting-extension`) as it is the artifact exposed to the end user.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few small suggestions.

**Review by gastaldi — APPROVED** [review]

File: docs/src/main/asciidoc/building-my-first-extension.adoc (hunk 9 above)
Comment (glefloch) [reply docs/src/main/asciidoc/building-my-first-extension.adoc:479]: yes sure I'm adding one.

**Review by gsmet — APPROVED** [review]

**Pull request merged**